src/main/drivers/accgyro_mpu.c

```diff
@@ -234,8 +234,7 @@ static bool detectSPISensorsAndUpdateDetectionResult(gyroDev_t *gyro, const sens
 #else
     UNUSED(bus);
 #endif
-    if (mpu6000SpiDetect(&gyro->bus)) {
-        gyro->mpuDetectionResult.sensor = MPU_60x0_SPI;
+    if ((gyro->mpuDetectionResult.sensor = mpu6000SpiDetect(&gyro->bus))) {
```

Comment:
I understand now. I had overlooked the sharing of 6500 code by other drivers. Frankly, the way this is done is a bit of a mess with little consistency. I've just added a fix for this, but at some point this should have a more thorough overhaul.

Can you check my latest change?